tools/xcodeproj_shims/install.sh

```diff
@@ -45,6 +45,14 @@ com.apple.product-type.application)
         "bazel-bin/$BAZEL_BIN_SUBDIR/${BAZEL_BUILD_TARGET_LABEL#*:}.runfiles/${BAZEL_BUILD_TARGET_WORKSPACE}/${BAZEL_BIN_SUBDIR}/$TARGET_NAME.zip"
     )
     ;;
+com.apple.product-type.app-extension)
+    input_options=(
+        "bazel-bin/$BAZEL_BIN_SUBDIR/${FULL_PRODUCT_NAME}"
+        "bazel-bin/$BAZEL_BIN_SUBDIR/${TARGET_NAME}_archive-root/$TARGET_NAME${WRAPPER_SUFFIX:-}"
+        "bazel-bin/$BAZEL_BIN_SUBDIR/${BAZEL_BUILD_TARGET_LABEL#*:}.runfiles/${BAZEL_BUILD_TARGET_WORKSPACE}/${BAZEL_BIN_SUBDIR}/${FULL_PRODUCT_NAME}"
+        "bazel-bin/$BAZEL_BIN_SUBDIR/${BAZEL_BUILD_TARGET_LABEL#*:}.runfiles/${BAZEL_BUILD_TARGET_WORKSPACE}/${BAZEL_BIN_SUBDIR}/$TARGET_NAME.zip"
```

Comment:
In practice I don't know when/if these are relevant locations and were mirrored from the `application` case above, and just put these in place discussion purposes in this PR

Reply:
If this works for your use case it seems like a fine addition to merge in, however it'd be ideal to have a sample/ integration test so we can ensure your use case is working in the future

The other thing I as wondering, is that you mentioned building the app as a dependency of this: if we go that approach, is there a reason to run Bazel 2 times for the extension if that extension inside of the app? If it's already built can't we just point Xcode at the app directory?

The last question I had is around the experience for extensions in general. If I have an app with `10` extensions and I'm iterating on do I need to constantly re-build, re-bundle, and re-link all of them? Does Xcode allow you to only build 1 app at a time or all 10?

Reply:
If the approach seems good, I'll go ahead and make a sample integration and test to show it working.

Re: Running Bazel twice – Though the 2nd "build" will obviously be pretty quick, I'm trying to think of how this would work to skip having to do that. Either:

1. Still build 2 Xcode build targets so each build target could manage its installation, but somehow script in "skipping" the Bazel build itself.
2. Build just 1 Xcode build target, but somehow infer all of the additional built products it needs to install in derived data

Since the install script relies heavily on Xcode build vars and individual target settings for the specific built product for that build target (i.e. `$TARGET_NAME`, `$FULL_PRODUCT_NAME`), I'm not sure what the options are here except `find *.app *.appex` in the bazelbin.

I wouldn't be surprised if there was a clever way to do this, but probably easier for others 😀

Reply:
> l. Say I have an app with 10 extensions. If I'm iterating on a single one do I need to constantly re-build, re-bundle, and re-link all of them?

Similar to how you'd do this in classic Xcode, you'd pick one of them (as in the activated scheme for that extension), and that scheme would build both the app target as well as that extension target.

The experience here is similar except this will clumsily run bazel twice, with the 2nd one being a (near) no-op.

Reply:
For reference, this is what an out-of-the-box app + share extension looks like in Xcode. Xcode automatically adds both build targets to the scheme for the extension itself.

![CleanShot 2021-07-02 at 13 37 34](https://user-images.githubusercontent.com/75236/124316106-aeab9a80-db3a-11eb-999d-67d45d613b7b.png)